lib/pages/AppDetails/AppDetails.tsx

```diff
@@ -138,6 +142,57 @@ const AppDetails: React.FC<AppDetailsProps> = ({
       })
     : []
 
+  // 4. 'More (Marketplace apps) from publisher' section logic
+
+  const [moreAppsFromPublisher, setMoreAppsFromPublisher] = useState([])
+
+  // Retrieves the 4 latest apps from the publisher
+  useEffect(() => {
+    if (retrievedSelectedAppDetails) {
+      getFilteredMarketplaceAppsAction({
+        org_id: [`${selectedAppDetails.orgId}`],
+        label: [],
+        sort_by: 'updated',
+        order: 'desc',
+        page: 1,
+        pageSize: 4,
+        search: '',
+      })
+    }
+  }, [retrievedSelectedAppDetails])
+
+  // Filters out the app in view if present, and reduces the retrieved apps to 3 if not
+  useEffect(() => {
+    if (retrievedFilteredMarketplaceApps) {
+      /* We only want the three most recently updated apps, and we want them in a format
+      that can be used by our App Catalog component. */
+      const appsFromPublisher = filteredMarketplaceApps
+        .filter((app) => {
+          return app.id !== selectedAppDetails.id
+        })
+        .slice(0, 3)
+        .map((app) => {
+          return {
+            appDescription:
+              app.shortDescription.length > 0
```

Comment:
This condition is used in 2 other places. Needs to be extracted and refactored (and possibly changed to non-ternary syntax).